mountpoint-s3-crt-sys/build.rs

```diff
@@ -301,6 +305,25 @@ fn compile_logging_shim(crt_include_dir: impl AsRef<Path>) {
     println!("cargo:rerun-if-changed=src/logging_shim.c");
 }
 
+/// Returns kind and modifiers to use while statically linking the given library.
+/// Should be used in `cargo:rustc-link-lib`.
+fn static_link_modifiers_for_lib(library_name: &str) -> &'static str {
+    // Up until v1.82, Rust was passing `+whole-archive` linker flag while building the tests.
+    // That's no longer the case with v1.82 and our tests started to fail with undefined references.
+    // To preserve this behaviour, we're passing `+whole-archive` explicitly for `aws-c-common`
+    // if we're doing a debug build.
+    // See https://github.com/rust-lang/rust/pull/128400.
+    // A debug build does not necessarily mean we're building the tests, but there is no way to
+    // detect if we're building the tests in `build.rs` at the moment, and this is the best approximation.
+    // See https://github.com/rust-lang/cargo/issues/1581.
+    let is_debug = get_env("PROFILE").unwrap_or_default() == "debug";
+    if is_debug && library_name == "aws-c-common" {
```

Comment:
Shall we simplify and always use `+whole-archive` for `aws-c-common`? Is the difference in final size on `release` significant?

Reply:
Seems like it doesn't affect the size of the `mount-s3` binary:
```bash
$ # with `+whole-archive`
$ ls -lah target/release/mount-s3
-rwxr-xr-x 2 .. .. 131M Oct 23 09:53 target/release/mount-s3

$ # without `+whole-archive`
$ ls -lah target/release/mount-s3
$ -rwxr-xr-x 2 .. .. 131M Oct 23 09:58 target/release/mount-s3
```

I updated to include `+whole-archive` always for `aws-c-common`.